v2/googlecloud-to-neo4j/src/test/java/com/google/cloud/teleport/v2/neo4j/templates/SinglePassGraphImportIT.java

```diff
@@ -0,0 +1,163 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.neo4j.templates;
+
+import static com.google.cloud.teleport.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+import static com.google.cloud.teleport.it.truthmatchers.PipelineAsserts.assertThatResult;
+
+import com.google.cloud.teleport.it.common.PipelineLauncher.LaunchConfig;
+import com.google.cloud.teleport.it.common.PipelineLauncher.LaunchInfo;
+import com.google.cloud.teleport.it.common.PipelineOperator.Result;
+import com.google.cloud.teleport.it.common.TestProperties;
+import com.google.cloud.teleport.it.common.utils.ResourceManagerUtils;
+import com.google.cloud.teleport.it.gcp.TemplateTestBase;
+import com.google.cloud.teleport.it.neo4j.Neo4jResourceManager;
+import com.google.cloud.teleport.it.neo4j.conditions.Neo4jQueryCheck;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(GoogleCloudToNeo4j.class)
+@RunWith(JUnit4.class)
+public class SinglePassGraphImportIT extends TemplateTestBase {
+  private Neo4jResourceManager neo4jClient;
+
+  @Before
+  public void setup() {
+    neo4jClient =
+        Neo4jResourceManager.builder(testName)
+            .setAdminPassword("letmein!")
+            .setHost(TestProperties.hostIp())
+            .build();
+  }
+
+  @After
+  public void tearDown() {
+    ResourceManagerUtils.cleanResources(neo4jClient);
+  }
+
+  @Test
+  public void importsGraphInSinglePassByMergingEdgeNodes() throws IOException {
+    String spec =
+        contentOf(
+            "/testing-specs/verbose-syntax/inline-json-bq-northwind-single-pass-jobspec.json");
+    gcsClient.createArtifact("inline-data-to-neo4j.json", spec);
+    gcsClient.createArtifact(
+        "neo4j-connection.json",
+        String.format(
+            "{\n"
+                + "  \"server_url\": \"%s\",\n"
+                + "  \"database\": \"%s\",\n"
+                + "  \"auth_type\": \"basic\",\n"
+                + "  \"username\": \"neo4j\",\n"
+                + "  \"pwd\": \"%s\"\n"
+                + "}",
```

Comment:
hopefully we can swap to Java >15 soon and use [text blocks](https://www.baeldung.com/java-text-blocks) :-)